pdi/src/expression/impl.cxx

```diff
@@ -80,7 +80,46 @@ string Expression::Impl::to_string(Context& ctx) const
 
 Ref Expression::Impl::to_ref(Context& ctx, Datatype_sptr type) const
 {
-	Ref_rw result{aligned_alloc(type->alignment(), type->buffersize()), [](void* v) { free(v); }, type, true, true};
+	// bug on MACOS with aligned_alloc
+	// auto data = std::aligned_alloc(type->alignment(), type->buffersize());
+	// Ref_rw result{data, [](void* v) { free(v); }, type, true, true};
+	// copy_value(ctx, result.get(), type);
+	// return result;
+
+	// === option 1: use hand-written version fo aligned_alloc
+	// size_t size = type->buffersize() + (type->alignment() - 1);
+	// void* buffer = operator new (size);
+	// void* data = std::align(type->alignment(), type->buffersize(), buffer, size);
+	// Ref_rw result{data, [buffer](void*) { operator delete (buffer); }, type, true, true};
+	// copy_value(ctx, result.get(), type);
+	// return result;
+
+	// === option 2: use posix_memalign
+	// void * data;
+	// int err = posix_memalign(&data, type->alignment()*sizeof(void*), type->buffersize());
+	// Ref_rw result{data, [](void* v) { free(v); }, type, true, true};
+	// copy_value(ctx, result.get(), type);
+	// return result;
+
+	// === option 3: use aligned_alloc, falls back to hand-written version if fails
+	// auto data = std::aligned_alloc(type->alignment(), type->buffersize());
+	// if(data) {
+	// 	Ref_rw result{data, [](void* v) { free(v); }, type, true, true};
+	// 	copy_value(ctx, result.get(), type);
+	// 	return result;
+	// }
+	// else {
+	// 	size_t size = type->buffersize() + (type->alignment() - 1);
+	// 	void* buffer = operator new (size);
+	// 	data = std::align(type->alignment(), type->buffersize(), buffer, size);
+	// 	Ref_rw result{data, [buffer](void*) { operator delete (buffer); }, type, true, true};
+	// 	copy_value(ctx, result.get(), type);
+	// 	return result;
+	// }
+
+	// option 4: use operator new with align_val_t (C++17)
+	auto data = operator new (type->buffersize(), static_cast<std::align_val_t>(type->alignment()));
+	Ref_rw result{data, [](void* v) { operator delete (v); }, type, true, true};
```

Comment:
I think you also have to specify the alignment during deletion, https://en.cppreference.com/w/cpp/memory/new/operator_delete